src/Google/GlobalSiteTag.php

```diff
@@ -155,13 +189,164 @@ public function maybe_display_event_snippet( string $ads_conversion_id, string $
 		$order->update_meta_data( self::ORDER_CONVERSION_META_KEY, 1 );
 		$order->save_meta_data();
 
-		printf(
-			'<script>gtag("event", "conversion", {"send_to": "%s","value": "%s","currency": "%s","transaction_id": "%s"});</script>',
+		$conversion_gtag_info =
+		sprintf(
+			'gtag("event", "conversion", {
+			send_to: "%s",
+			value: %f,
+			currency: "%s",
+			transaction_id: "%s"});',
 			esc_js( "{$ads_conversion_id}/{$ads_conversion_label}" ),
-			esc_js( $order->get_total() ),
+			$order->get_total(),
 			esc_js( $order->get_currency() ),
 			esc_js( $order->get_id() ),
 		);
+		wp_print_inline_script_tag( $conversion_gtag_info );
+
+		// Get the item info in the order
+		$item_info = [];
+		foreach ( $order->get_items() as $item_id => $item ) {
+			$product_id   = $item->get_product_id();
+			$product_name = $item->get_name();
+			$quantity     = $item->get_quantity();
+			$price        = $item->get_subtotal();
+			$item_info [] = sprintf(
+				'{
+				id: "gla_%s",
+				price: %f,
+				google_business_vertical: "retail",
+				name: "%s",
+				quantity: %d,
+				}',
+				esc_js( $product_id ),
+				$price,
+				esc_js( $product_name ),
+				$quantity,
+			);
+		}
+
+		// Check if this is the first time customer
+		$is_new_customer = $this->is_first_time_customer( $order->get_billing_email() );
+
+		// Track the purchase page
+		$language = $this->wp->get_locale();
+		if ( 'en_US' === $language ) {
+			$language = 'English';
+		}
+		$purchase_page_gtag =
+		sprintf(
+			'gtag("event", "purchase", {
+			ecomm_pagetype: "purchase",
+			send_to: "%s",
+			transaction_id: "%s",
+			currency: "%s",
+			country: "%s",
+			value: %f,
+			new_customer: %s,
+			tax: "%s",
```

Comment:
Tax is also a monetary amount like value.

Reply:
Done.